Communitter/src/main/java/com/communitter/api/util/BasicAuthorizationUtil.java

```diff
@@ -0,0 +1,32 @@
+package com.communitter.api.util;
+
+import com.communitter.api.exception.NotAuthorizedException;
+import com.communitter.api.model.User;
+import com.communitter.api.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class BasicAuthorizationUtil {
```

Comment:
authenticated userı @AuthenticationPrincipal annotation ı ile aşağıdaki şekilde direk alabiliyoruz. Ekipçe değerlendirebiliriz. 
@GetMapping("/user")
    public String getUser(@AuthenticationPrincipal UserDetails userDetails) {
        return "User Details: " + userDetails.getUsername();
    }

Reply:
@ynsmrkrkmz bunun su anki util de kullanilan functiondan farki ne olacak peki?

Reply:
detay olarak linkte detaylı açıklama var. https://backendhance.com/en/blog/2023/authentication-principal/ 


You get easy access to user information without directly using the SecurityContext.
You increase the testability of your code, as you can mock the controller methods more easily.
You adhere to the Single Responsibility Principle and avoid casting in the API layer.

Reply:
@ynsmrkrkmz bunu kullanacaksak bir refactoreing yapmamiz gerekiyor projede ilerlemeden. Ben cok essential oldugunu dusunmuyorum bu arada. Yine asap konusunu acalim bu PR'da beklemesin.